.github/workflows/compile.yaml

```diff
@@ -0,0 +1,75 @@
+name: compile
+
+on: push
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  compile:
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler: [ 'g++', 'clang' ]
+
+    runs-on: ubuntu-latest
+    container:
+      image: debian:testing
+
+    defaults:
+      run:
+        shell: sh
+
+    steps:
+    - name: Install dependencies
+      run: |
+        apt-get -y update
+        apt-get -y install --no-install-recommends ${{ matrix.compiler }} lld meson pkgconf git ca-certificates locales-all
+
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Configure Meson
+      run: |
+        if [ ${{ matrix.compiler }} = g++ ]; then linker=gold; else linker=lld; fi
+        CXX_LD=$linker meson setup build --buildtype=debug -Dcompile_library=true -Dpedantic=true -Dbuild_tests=true -Dbuild_examples=true -Dgenerate_cmake_config=false -Db_lto=false -Dasan_examples=true
+
+    - name: Build
+      run: meson compile -C build
+
+    - name: Test
+      run: meson test -C build --verbose
+
+  windows:
+    runs-on: windows-2022
+    # Containers could be used even on Windows, see
+    # https://github.com/microsoft/lightgbm-benchmark/commit/f4c79cb49b72fc9fefa677182855811a2eb45554
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    # It is necessary to remove MinGW and StrawberryPerl as they both provide
+    # GCC, and Meson prefers to use that instead of MSVC.
+    - name: Install dependencies
+      run: |
+        choco uninstall mingw strawberryperl --yes --all-versions --remove-dependencies --skip-autouninstaller --no-color
+        Remove-Item -Path C:\Strawberry -Recurse
+        choco install pkgconfiglite --yes --skip-virus-check --no-color
+        pip install meson ninja
+
+    - uses: ilammy/msvc-dev-cmd@v1
+
+    - name: Configure Meson
+      run: meson setup build --buildtype=debug -Dcompile_library=true -Dpedantic=true -Dbuild_tests=false -Dbuild_examples=true -Dgenerate_cmake_config=false -Db_lto=false
+      env:
+        CPPFLAGS: -DTOML_EXPORTED_CLASS=TOML_EXPORTED_FUNCTION
```

Comment:
I don't think this will achieve anything; in the header file there's some logic that pre-emptively #undefs them both when `TOML_SHARED_LIB` is in use, and sets them manually

The rationale is that in windows DLL builds, you can `dllexport` a whole class, or `dllexport` specific functions _in_ a class, but not both

Reply:
Oh ok, I was just trying out random stuff 😅️